### PR TITLE
Chore: Allow http to set custom headers

### PR DESCRIPTION
**Pull request by RommHui**

### CODE REVIEW

**Comment by RommHui** [comment]
An exmaple configuration can be:
`
proxies:
    - name: example
      type: http
      server: 127.0.0.1
      port: 443
      headers:
        User-Agent: chrome
        X-Forwarded-For: 8.8.8.8
`

**Comment by Dreamacro** [comment]
Can you explain what specific custom header solves your problem?

**Comment by RommHui** [comment]
@Dreamacro Some http proxies need to verify certain specific request headers, but the original http proxy of Clash does not support custom request headers, which causes the http proxy to fail to operate normally

**Comment by Dreamacro** [comment]
@RommHui For example? I want to see a practical example.

File: adapter/outbound/http.go [hunk 1]
```diff
@@ -134,6 +136,11 @@ func NewHttp(option HttpOption) *Http {
 		}
 	}
 
+	headers := make(http.Header)
```
Comment (Dreamacro) [inline adapter/outbound/http.go:139]: perfer `http.Header{}` in this repo

**Review by Dreamacro — CHANGES_REQUESTED** [review]

File: adapter/outbound/http.go (hunk 1 above)
Comment (RommHui) [reply adapter/outbound/http.go]: `make(http.Header)` has been changed to `http.Header{}`

**Comment by Dreamacro** [comment]
Thanks for contributing

**Pull request merged**